### PR TITLE
Add integers 0.5.0.

### PR DESCRIPTION
**Pull request by yallop**

### CODE REVIEW

File: packages/integers/integers.0.5.0/opam [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+license: "MIT"
+
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+
+depends: [
+  "ocaml" {>= "4.02"}
```
Comment (kit-ty-kate) [inline packages/integers/integers.0.5.0/opam:19]:
```suggestion
  "ocaml" {>= "4.05"}
```
The build fails on 4.04 and older with:
```
#=== ERROR while compiling integers.0.5.0 =====================================#
# context              2.0.9 | linux/x86_64 | ocaml-base-compiler.4.02.3 | pinned(https://github.com/ocamllabs/ocaml-integers/archive/0.5.0.tar.gz)
# path                 ~/.opam/4.02/.opam-switch/build/integers.0.5.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p integers -j 31
# exit-code            1
# env-file             ~/.opam/log/integers-2463-865888.env
# output-file          ~/.opam/log/integers-2463-865888.out
### output ###
#       ocamlc src/.integers.objs/byte/signed.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.02/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.integers.objs/byte -intf-suffix .ml -no-alias-deps -o src/.integers.objs/byte/signed.cmo -c -impl src/signed.ml)
# File "src/signed.ml", line 90, characters 24-41:
# Error: Unbound value int_of_string_opt
#     ocamlopt src/.integers.objs/native/signed.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.02/bin/ocamlopt.opt -w -40 -g -I src/.integers.objs/byte -I src/.integers.objs/native -intf-suffix .ml -no-alias-deps -o src/.integers.objs/native/signed.cmx -c -impl src/signed.ml)
# File "src/signed.ml", line 90, characters 24-41:
# Error: Unbound value int_of_string_opt
```

**Comment by kit-ty-kate** [comment]
Also it looks like many packages depending on it are failing. Could you have a look at them?

**Comment by yallop** [comment]
@fdopen: c3d63db adds version constraints to `ppx_cstubs`, since the extended interface in this release of `integers` breaks the `ppx_cstubs` build`

**Comment by avsm** [comment]
Looks like the pkcs11 packages also fail with this. Copying @emillon in case he's still looking at those as he is listed as maintainer.

**Comment by yallop** [comment]
I've added constraints to the pkcs11 packages (and proposed a fix upstream here: https://github.com/cryptosense/pkcs11/pull/122).

**Comment by avsm** [comment]
Many thanks for taking the time to add all the constraints, @yallop! This looks fine to merge now. You may want to announce this on <https://discuss.ocaml.org>, where we have a Community category and an `announce` tag for this purpose.

**Comment by emillon** [comment]
> Looks like the pkcs11 packages also fail with this. Copying @emillon in case he's still looking at those as he is listed as maintainer.

Ping @bbc2 who'll be more able to fix this :)

**Comment by bbc2** [comment]
Hi @emillon o/  Thanks for letting me know.  We'll be looking into this. And thank you @yallop  for the adjusted constraints and upstream PR.

**Pull request merged**